2001-08-27  Miles Bader  <miles@gnu.org>

	* simops.c (OP_10007E0): Don't subtract 4 from the new trap PC.

diff -up sim/v850/simops.c.\~2\~ sim/v850/simops.c
--- sim/v850/simops.c.~2~	Fri Aug 24 15:38:19 2001
+++ sim/v850/simops.c	Mon Aug 27 17:34:08 2001
@@ -1875,7 +1875,7 @@ OP_10007E0 ()
       ECR |= 0x40 + OP[0];
       /* Flag that we are now doing exception processing.  */
       PSW |= PSW_EP | PSW_ID;
-      PC = ((OP[0] < 0x10) ? 0x40 : 0x50) - 4;
+      PC = (OP[0] < 0x10) ? 0x40 : 0x50;
 
       return 0;
     }
